tempesta_fw/t/functional/regression/test_reboot.py

```diff
@@ -0,0 +1,82 @@
+"""
+Test TempestaFW reeboot under heavy load.
+"""
+
+from __future__ import print_function
+import unittest
+import threading
+from helpers import deproxy, tf_cfg, tempesta, remote, control
+from testers import stress
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2017 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+class RebootTest(stress.StressTest):
+    """Start load only when TempestaFW is up"""
+
+    restart_cycles = 10
+
+    # Override test duration. Mostly there is no sence to load TempestaFW for a
+    # long time before rebooting it.
+    duration = 1
+
+    def create_clients(self):
+        self.wrk = control.Wrk()
+        self.wrk.duration = self.duration + 1
+        self.clients = [self.wrk]
+
+    def reboot_routine(self, config):
+        # Set defconfig for Tempesta.
+        self.tempesta.config.set_defconfig(config)
+        self.configure_tempesta()
+        control.servers_start(self.servers)
+        self.wrk.prepare()
+
+        for i in range(self.restart_cycles):
+            self.tempesta.start()
+            threading.Timer(self.duration, self.tempesta.stop).start ()
+            control.client_run_blocking(self.wrk)
```

Comment:
It seems you restart wrk on each iteration. Giving that there is only 1 second delay, we won't have significant queues sizes, so probably the conditions are still relatively mercy. I'd suggest to run heavy wrk, wait few seconds, and start the restarting loop with random delay in range [1..20] seconds.